pkg/controller/machine.go

```diff
@@ -223,22 +223,18 @@ func (c *Controller) updateMachine(machine *machinev1alpha1.Machine) (*machinev1
 	return c.machineClient.MachineV1alpha1().Machines().Update(machine)
 }
 
-func (c *Controller) clearMachineErrorIfSet(machine *machinev1alpha1.Machine, reason machinev1alpha1.MachineStatusError) (*machinev1alpha1.Machine, error) {
-	if machine.Status.ErrorReason != nil && *machine.Status.ErrorReason == reason {
-		machine.Status.ErrorMessage = nil
-		machine.Status.ErrorReason = nil
-		return c.updateMachine(machine)
-	}
-	return machine, nil
+func (c *Controller) clearMachineErrorIfSet(machine *machinev1alpha1.Machine) (*machinev1alpha1.Machine, error) {
+	machine.Status.ErrorMessage = nil
+	machine.Status.ErrorReason = nil
+	return c.updateMachine(machine)
 }
 
+// updateMachine updates machine's ErrorMessage and ErrorReason regardless if they were set or not.
+// This essentially overwrites previous values.
 func (c *Controller) updateMachineError(machine *machinev1alpha1.Machine, reason machinev1alpha1.MachineStatusError, message string) (*machinev1alpha1.Machine, error) {
```

Comment:
those errors are not permanent e.g. a user can provide incorrect credentials that would result in setting a `terminal` error during creation but later on a user may wish to delete machine that would result in a `terminal` error during deletion.